### ISSUE TITLE
Mismatch in hashicorp/google provider constraints between `google_redis_instance` and `enable_apis` modules

### ISSUE THREAD

**Issue by aprettyloner** [terraform-google-modules/terraform-google-memorystore#236]
### TL;DR

The 11.0.0 release (https://github.com/terraform-google-modules/terraform-google-memorystore/pull/230) involved a major hashicorp/google provider bump to v6. However, this only applies to the `google_redis_instance` resource. The `enable_apis` module still has a `< 6.0.0` constraint which is incompatible.

This will be fixed with a pre-existing PR: https://github.com/terraform-google-modules/terraform-google-memorystore/pull/234

### Expected behavior

We expect that bumping to `11.0.0` would not cause provider constraint issues

<img width="445" alt="image" src="https://github.com/user-attachments/assets/a8b8b968-cd8d-421e-a6dd-3e6f133e7794">

### Observed behavior

```
Downloading registry.terraform.io/terraform-google-modules/memorystore/google 11.0.0 for service_stu3obs_evidence_mapper.redis_instance.redis_instance...
...
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider hashicorp/google: no available releases match the given constraints
│ >= 3.43.0, >= 4.74.0, >= 5.27.0, < 6.0.0, >= 6.0.1, < 7.0.0
```

This discrepancy is coming from 
```
    ├── module.redis_instance
        └── module.redis_instance
            ├── provider[registry.terraform.io/hashicorp/google] >= 4.74.0, < 7.0.0
            └── module.enable_apis
                ├── provider[registry.terraform.io/hashicorp/google] >= 3.43.0, < 6.0.0
                └── provider[registry.terraform.io/hashicorp/google-beta] >= 3.43.0, < 6.0.0
```

### Terraform Configuration

We have a very light wrapper over the `terraform-google-memorystore` module, invoked as such:

```hcl
module "redis_instance" {
  source  = "app.terraform.io/virtahealth/redis-instance/google"
  version = "1.3.0"

  count        = var.config.redis.enabled ? 1 : 0
  project      = var.project
  name         = "${var.name}-redis"
  display_name = "Redis for the ${var.name} service"

  auth_enabled          = local.redis_auth_enabled
  authorized_network_id = data.google_compute_network.vpc.id
  connect_mode          = local.redis_connect_mode
  customer_managed_key  = local.redis_customer_managed_key
  enable_apis           = local.redis_enable_apis
  labels                = local.labels
  memory_size_gb        = local.redis_memory_size_gb
  persistence_config = {
    persistence_mode    = local.redis_persistence_mode
    rdb_snapshot_period = local.redis_rdb_snapshot_period
  }
  read_replicas_mode      = local.redis_read_replicas_mode
  redis_configs           = local.redis_configs
  redis_version           = local.redis_version
  region                  = local.redis_region
  replica_count           = local.redis_replica_count
  reserved_ip_range       = local.redis_reserved_ip_range
  transit_encryption_mode = local.redis_transit_encryption_mode
}
```

Our internal module calls the current `memorystore` module. This issue arose when bumping to `11.0.0`.

```hcl

module "redis_instance" {
  source  = "terraform-google-modules/memorystore/google"
  version = "8.0.0" # https://github.com/terraform-google-modules/terraform-google-memorystore/releases
...
```

### Terraform Version

```sh
Terraform v1.9.0
```

### Additional information

Simple fix: Merge https://github.com/terraform-google-modules/terraform-google-memorystore/pull/234
